web/src/input_pill.ts

```diff
@@ -148,29 +151,32 @@ export function create<T>(opts: InputPillCreateOptions<T>): InputPillContainer<T
                 blueslip.error("no type defined for the item");
                 return;
             }
+            let pill_html;
+            if (item.type === "search_user") {
+                pill_html = render_search_user_pill(item);
```

Comment:
It feels a bit weird to have this custom logic here, but that's sort of a consequence of sharing a lot of code with `input_pill` (that's encapsulated in here) and also wanting to do a lot of custom things.

Reply:
It's not obvious to me that `render_input_pill` couldn't be adapted to handle this, but I think this is fine and it's best to try to do further refactoring around that interface in a follow-up PR, that perhaps tries to use object inheritance.

Reply:
I'll add that to a list of followups :)